### PR TITLE
Merging alpha/4.0.0 into master, next release being 4.0.0

### PR DESCRIPTION
**Pull request by wilbrdt**
## Purpose

It is hard to keep the alpha/v4.0.0 branch following master while adding new
features.

## Proposal

As we hopefully approach the release of v4.0.0, we've decided to merge alpha
into master.
We can still make hotfixes on older versions with branches created from latest v3.9.0 tag.

### CODE REVIEW

**Review by SergioSim — APPROVED** [review]
LGTM :+1: 

image

File: UPGRADE.md [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+
+### 3.x.x to 4.0.0
+
+#### Upgrade history syntax
+
+CLI syntax has been changed from `fetch` & `push` to `read` & `write` affecting the command history. You must replace the command history after updating:
+- locate your history file path, which is in `{ RALPH_APP_DIR }/history.json` (defaults to `.ralph/history.json`)
+- run the commands below to update history
+```
+$ sed -i 's/"fetch"/"read"/g' { my_history_file_path }
```
[the remark this answers is not in the document]
Comment (Leobouloc) [reply UPGRADE.md:15]: Yeah... Given the current user base, and the fact that it seems like a very bad choice for a name, the risk seems to me low enough to take.

[the remark this answers is not in the document]
Comment (wilbrdt) [reply UPGRADE.md:15]: Thought about it to, and I too think the risk is low enough!

**Review by Leobouloc — COMMENTED** [review]
We should add the changes made to `auth.json` to the `UPGRADE.md` file: 
https://github.com/openfun/ralph/pull/404 (waiting to be merged)

Other than that, it looks fine to me !

**Review by Leobouloc — APPROVED** [review]

**Pull request merged**